src/js/components/JobForm.js

```diff
@@ -43,21 +37,60 @@ class JobForm extends SchemaForm {
       return;
     }
 
-    this.validateForm();
-    // Provide updated model to get new multipleDefinition
-    let model = this.triggerTabFormSubmit();
-    this.multipleDefinition = this.getNewDefinition(model);
-    this.props.onChange(this.getDataTriple(model));
-    // Only update if schedule changed
-    if (scheduleEnabledChange) {
+    // Fetch the updated model from the <Form /> component
+    // nested deeper in the component tree.
+    this.model = this.triggerTabFormSubmit();
```

Comment:
Besides, even if I don't set the `.model` I will need to set the `.multipleDefinition`, because I want the validation to happen outside the `handleFormChange` event. Therefore since we are already violating the principle, I thought of keeping it more compatible with the whatever is the weird logic of `SchemaForm`